store/store.go

```diff
@@ -2,108 +2,113 @@
 package store
 
 import (
-	"context"
-	"time"
+	"errors"
+	"os"
+	"sync"
 
-	"github.com/gluster/glusterd2/gdctx"
+	"github.com/gluster/glusterd2/pkg/elasticetcd"
 
-	log "github.com/Sirupsen/logrus"
 	"github.com/coreos/etcd/clientv3"
 	"github.com/coreos/etcd/clientv3/concurrency"
-	"github.com/pborman/uuid"
-	config "github.com/spf13/viper"
 )
 
 const (
 	// GlusterPrefix prefixes all paths in the store
-	GlusterPrefix     = "gluster/"
-	sessionTTL        = 30 // used for etcd mutexes and liveness key
-	livenessKeyPrefix = GlusterPrefix + "alive/"
+	GlusterPrefix = "gluster/"
+	sessionTTL    = 30 // used for etcd mutexes and liveness key
 )
 
-// Store variable can be imported by packages which need access to the store
-var Store *GDStore
+var (
+	// Store is the default GDStore that must to be used by the packages in GD2
+	Store *GDStore
```

Comment:
We may also need mechanism (global lock or graceful server shutdown) to stop accepting new client connections (HTTP and sunrpc) during a store restart (add/remove peer). The goroutines that access the store will be serving client requests.